src/components/ApplicationLauncher/ApplicationLauncher.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import ApplicationLauncherToggle from './ApplicationLauncherToggle';
+import { Dropdown } from '../Dropdown';
+
+class ApplicationLauncher extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showLauncher: false,
+      type: props.type
```

Comment:
why add type to state rather than simply using the prop? It should only be mutable by the caller